pkg/plugin/query.go

```diff
@@ -4,27 +4,52 @@ import (
 	"fmt"
 	"net/url"
 	"path"
+	"regexp"
+	"sort"
 	"strconv"
+	"strings"
 	"time"
 
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+
 	"github.com/VictoriaMetrics/victorialogs-datasource/pkg/utils"
 )
 
 const (
-	instantQueryPath = "/select/logsql/query"
-	tailQueryPath    = "/select/logsql/tail"
-	defaultMaxLines  = 1000
+	instantQueryPath    = "/select/logsql/query"
+	tailQueryPath       = "/select/logsql/tail"
+	statsQueryPath      = "/select/logsql/stats_query"
+	statsQueryRangePath = "/select/logsql/stats_query_range"
+	defaultMaxLines     = 1000
+	legendFormatAuto    = "__auto"
+	metricsName         = "__name__"
+	defaultInterval     = 15 * time.Second
+)
+
+// QueryType represents query type
+type QueryType string
+
+const (
+	QueryTypeInstant    QueryType = "instant"
+	QueryTypeStats      QueryType = "stats"
+	QueryTypeStatsRange QueryType = "statsRange"
 )
 
 // Query represents backend query object
 type Query struct {
-	RefID        string `json:"refId"`
-	Expr         string `json:"expr"`
-	LegendFormat string `json:"legendFormat"`
-	MaxLines     int    `json:"maxLines"`
-	TimeRange    TimeRange
-	IntervalMs   int `json:"intervalMs"`
-	url          *url.URL
+	RefID           string `json:"refId"`
```

Comment:
Seems like most of the fields here are just `backend.DataQuery` fields. Should we embedd this struct instead?

Reply:
fixed